src/downloadPubchemInfo.py

```diff
@@ -0,0 +1,108 @@
+# Porthmeus
+# 15.02.23
+
+
+'''
+Get a list of metabolite names, search pubchem for it and retrieve an InChI/Smile string from it.
+'''
+
+
+import pubchempy as pcp
+import pandas as pd
+import numpy as np
+import time
+import sys
+
+# global variable for the atoms list
+atoms_list = ['H', 'He', 'Li', 'Be', 'B', 'C', 'N', 'O', 'F', 'Ne', 'Na', 'Mg', 'Al', 'Si', 'P', 'S', 'Cl', 'Ar', 'K', 'Ca', 'Sc', 'Ti', 'V', 'Cr', 'Mn', 'Fe', 'Co', 'Ni', 'Cu', 'Zn', 'Ga', 'Ge', 'As', 'Se', 'Br', 'Kr', 'Rb', 'Sr', 'Y', 'Zr', 'Nb', 'Mo', 'Tc', 'Ru', 'Rh', 'Pd', 'Ag', 'Cd', 'In', 'Sn', 'Sb', 'Te', 'I', 'Xe', 'Cs', 'Ba', 'La', 'Ce', 'Pr', 'Nd', 'Pm', 'Sm', 'Eu', 'Gd', 'Tb', 'Dy', 'Ho', 'Er', 'Tm', 'Yb', 'Lu', 'Hf', 'Ta', 'W', 'Re', 'Os', 'Ir', 'Pt', 'Au', 'Hg', 'Tl', 'Pb', 'Bi', 'Po', 'At', 'Rn', 'Fr', 'Ra', 'Ac', 'Th', 'Pa', 'U', 'Np', 'Pu', 'Am', 'Cm', 'Bk', 'Cf', 'Es', 'Fm', 'Md', 'No', 'Lr', 'Rf', 'Db', 'Sg', 'Bh', 'Hs', 'Mt', 'Ds', 'Rg', 'Cn', 'Nh', 'Fl', 'Mc', 'Lv', 'Ts', 'Og', 'R']
+
+
+def downloadElements() -> pd.DataFrame:
+    '''
+    Download the elements table from pubchem and return it as a data frame
+    '''
+
+    # download the element list from pubchem
+    elements = pd.read_csv("https://pubchem.ncbi.nlm.nih.gov/rest/pug/periodictable/CSV?response_type=save&response_basename=PubChemElements_all")
+    return(elements)
+
+def looksLikeSumFormula(string:str, atoms:list = atoms_list) -> bool:
+    '''
+    Takes a string and tries to determine if it is likely a sum formula
```

Comment:
This will fail for stuff like C2H4O2Zn+2 but I assume we only have a subset of possible chemical formulas to check?

Reply:
Is there some definition of the Pubchem chemical formulas? Maybe a Backus-Naur-Form

Reply:
Yes there is - however not all users comply to it. There is a good chance that the code will fail with common words already, thus I see the need to improve here maybe. However, currently it is used only during a fallback if pubchem can not find an entry to the name given - than I try to remove these formulas, if they are part of the name. I will for now, leave the code as it is and put it as an improvement to the issues.

Reply:
Backus-Naur would look something like [<Atom>|<NumberOfTime>]n